subsys/bluetooth/controller/ll_sw/ull_adv_iso.c

```diff
@@ -307,55 +501,131 @@ static uint32_t ull_adv_iso_start(struct ll_adv_iso *adv_iso,
 
 	ret_cb = TICKER_STATUS_BUSY;
 	ret = ticker_start(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_ULL_HIGH,
-			   (TICKER_ID_ADV_ISO_BASE + adv_iso->bis_handle),
+			   (TICKER_ID_ADV_ISO_BASE + adv_iso->lll.handle),
 			   ticks_anchor, 0,
 			   HAL_TICKER_US_TO_TICKS(iso_interval_us),
 			   HAL_TICKER_REMAINDER(iso_interval_us),
 			   TICKER_NULL_LAZY,
-			   (ll_adv_iso->evt.ticks_slot + ticks_slot_overhead),
-			   ticker_cb, ll_adv_iso,
+			   (adv_iso->evt.ticks_slot + ticks_slot_overhead),
+			   ticker_cb, adv_iso,
 			   ull_ticker_status_give, (void *)&ret_cb);
 	ret = ull_ticker_status_take(ret, &ret_cb);
 
-	return 0;
+	return ret;
 }
 
-static inline struct ll_adv_iso *ull_adv_iso_get(uint8_t handle)
+static void mfy_iso_offset_get(void *param)
 {
-	if (handle >= CONFIG_BT_CTLR_ADV_SET) {
-		return NULL;
-	}
+	struct ll_adv_sync_set *sync;
+	struct lll_adv_iso *lll_iso;
+	uint32_t ticks_to_expire;
+	struct pdu_big_info *bi;
+	uint32_t ticks_current;
+	struct pdu_adv *pdu;
+	uint8_t ticker_id;
+	uint8_t retry;
+	uint8_t id;
+
+	sync = param;
+	lll_iso = sync->lll.iso;
+	ticker_id = TICKER_ID_ADV_ISO_BASE + lll_iso->handle;
+
+	id = TICKER_NULL;
+	ticks_to_expire = 0U;
+	ticks_current = 0U;
+	retry = 4U;
+	do {
+		uint32_t volatile ret_cb;
+		uint32_t ticks_previous;
+		uint32_t ret;
+
+		ticks_previous = ticks_current;
+
+		ret_cb = TICKER_STATUS_BUSY;
+		ret = ticker_next_slot_get(TICKER_INSTANCE_ID_CTLR,
+					   TICKER_USER_ID_ULL_LOW,
+					   &id,
+					   &ticks_current, &ticks_to_expire,
+					   ticker_op_cb, (void *)&ret_cb);
+		if (ret == TICKER_STATUS_BUSY) {
+			while (ret_cb == TICKER_STATUS_BUSY) {
+				ticker_job_sched(TICKER_INSTANCE_ID_CTLR,
+						 TICKER_USER_ID_ULL_LOW);
+			}
+		}
 
-	return &ll_adv_iso[handle];
+		LL_ASSERT(ret_cb == TICKER_STATUS_SUCCESS);
+
+		LL_ASSERT((ticks_current == ticks_previous) || retry--);
+
+		LL_ASSERT(id != TICKER_NULL);
+	} while (id != ticker_id);
+
+	pdu = lll_adv_sync_data_latest_peek(&sync->lll);
+	bi = big_info_get(pdu);
+	big_info_offset_fill(bi, ticks_to_expire, 0);
+	/* FIXME: Fill the payload count */
 }
 
-static int init_reset(void)
+static inline struct pdu_big_info *big_info_get(struct pdu_adv *pdu)
 {
-	/* Initialize pool. */
-	mem_init(ll_adv_iso, sizeof(struct ll_adv_iso),
-		 sizeof(ll_adv_iso) / sizeof(struct ll_adv_iso), &adv_iso_free);
+	struct pdu_adv_com_ext_adv *p;
+	struct pdu_adv_ext_hdr *h;
+	uint8_t *ptr;
 
-	return 0;
+	p = (void *)&pdu->adv_ext_ind;
+	h = (void *)p->ext_hdr_adv_data;
+	ptr = h->data;
+
+	/* FIXME: CTEInfo */
+
+	if (h->aux_ptr) {
+		ptr += sizeof(struct pdu_adv_aux_ptr);
+	}
+
+	if (h->tx_pwr) {
+		ptr++;
+	}
+
+	/* Length encoded AD Format */
+	ptr += 2;
+
+	return (void *)ptr;
 }
 
+static inline void big_info_offset_fill(struct pdu_big_info *bi,
+					uint32_t ticks_offset,
+					uint32_t start_us)
+{
+	uint32_t offs;
+
+	offs = HAL_TICKER_TICKS_TO_US(ticks_offset) - start_us;
+	offs = offs / OFFS_UNIT_30_US;
+	if (!!(offs >> 13)) {
```

Comment:
nit: coerce-to-boolean strictly not needed here, as if(cond) already check for non-zeroness of cond.